commerce_coordinator/apps/commercetools/clients.py

```diff
@@ -1636,3 +1636,35 @@ def find_order_with_unprocessed_return_for_payment(
                 f"for payment: {payment_id} for customer: {customer_id}",
             )
             raise err
+
+    def get_order_by_payment_id(self, payment_id: str) -> Order:
+        """
+        Retrieves the order associated with a given payment ID.
+
+        Args:
+            payment_id (str): Payment ID
+
+        Returns:
+            Order if found.
+
+        Raises:
+            ValueError: If no order is found for the given payment ID.
+        """
+        try:
+            response = self.base_client.orders.query(
+                where=["paymentInfo(payments(id=:payment_id))"],
+                predicate_var={"payment_id": payment_id},
+            )
+
+            if not response or not response.results:
+                raise Exception(f"No order found for payment ID {payment_id}")
```

Comment:
```suggestion
                raise CommercetoolsError(f"No order found for payment ID {payment_id}")
```

Reply:
Well, in that case, the Exception class won't be watched for retrying according to the current retry criteria.